language/vm/vm-runtime/src/chain_state.rs

```diff
@@ -33,6 +33,12 @@ pub trait ChainState {
     // organized as a tree of GlobalRefs.
     /// Get a mutable reference to a resource stored on chain.
     fn load_data(&mut self, ap: &AccessPath, def: StructDef) -> VMResult<&mut GlobalRef>;
+
+    /// Get the serialized format of a `CompiledModule` from chain given a `ModuleId`.
```

Comment:
This seems to be a pretty good temporary step for us. Once we can have a better story for load_data, we can reduce `load_data` and `load_module` into `load(&self, access_path: &AccessPath) -> VMResult<Vec<u8>>`, which will make the api cleaner in the future.

Reply:
yeah my goal here is to minimize the number of traits we have and later (particularly as you progress with your work) break things up as needed. So as you noted I shoveled everything here for now ;) and agree that is a good way to move forward pending a more intelligent reorganization.
My recommendation is to minimize the number of traits and then look into what we have and work accordingly.
I think the `InterpreterContext` and `ChainState` are way too much already and again I think your work may clarify that API better